pkg/event/processor.go

```diff
@@ -66,7 +66,9 @@ const DefaultEventFlushInterval = 30 * time.Second
 
 const maxFlushWorkers = 1
 
-var pLogger = logging.GetLogger("EventProcessor")
+func pLogger(c context.Context) logging.OptimizelyLogProducer {
+	return logging.GetLogger(c, "EventProcessor")
+}
```

Comment:
Similar comment here, logger should be a field on the `BatchEventProcessor`

Reply:
Looks like I lost my original comment somewhere, so to clarify I think we should be consistent and store the logger on the structs and not look them up dynamically with each log statement. This adds unnecessary overhead to each log statement.